### PR TITLE
add module for dragonflye

### LINKED ISSUE #632: new module: Dragonflye
I'll be working on adding a module recipe for [Dragonflye](https://github.com/rpetit3/dragonflye) a fork of Shovill that supports ONT reads.

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by rpetit3**
This PR adds a module for [Dragonflye](https://github.com/rpetit3/dragonflye) a fork of Shovill that supports ONT reads.

## PR checklist

Closes #632 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by rpetit3** [comment]
This is going to fail due to MD5s. I submitted the PR and will see if I can get setup to test locally

**Comment by rpetit3** [comment]
WIP in progress, found a bug that I fixed https://github.com/rpetit3/dragonflye/commit/5e83b86693bcb3110d320bb7cb158d1ba0d660b9, once next version is on Bioconda I'll finish this

**Comment by rpetit3** [comment]
A few things.

1. had to use Nanopore data from bacass test-data, due to SC2 genome being too small
2. `flye` testing was disabled due to it exceeding 3gb of memory

File: modules/dragonflye/main.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    tuple val(meta), path("{miniasm,raven}-unpolished.gfa"), optional:true     , emit: gfa
+    tuple val(meta), path("flye-info.txt"), optional:true                      , emit: txt
+    path "*.version.txt"                                                       , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def memory = task.memory.toGiga()
+    """
+    dragonflye \\
+        --reads ${reads[0]} \\
```
Comment (grst) [inline modules/dragonflye/main.nf:37]: why `${reads[0]}`? Shouldn't it just fail with paired-end data?

File: modules/dragonflye/meta.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - reads:
+        type: file
+        description: Input Nanopore FASTQ file
```
Comment (grst) [inline modules/dragonflye/meta.yml:23]: pls add `pattern`

File: modules/dragonflye/main.nf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+options        = initOptions(params.options)
+
+process DRAGONFLYE {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::dragonflye=1.0.2" : null)
```
Comment (grst) [inline modules/dragonflye/main.nf:14]: conda version does not match the containers

**Review by grst — COMMENTED** [review]
Looks mostly good already! 
What about the conda tests? It seems it's not just mismatching checksums but fails completely.

**Comment by rpetit3** [comment]
@grst Thank you! I'll work on this in the next day or so

File: modules/dragonflye/meta.yml (hunk 2 above)
Comment (rpetit3) [reply modules/dragonflye/meta.yml:23]: added patter for input fastqs

**Comment by rpetit3** [comment]
@grst I think its all set!

File: modules/dragonflye/main.nf (hunk 1 above)
Comment (rpetit3) [reply modules/dragonflye/main.nf]: Honestly, because I used the Shovill module as template for this one, since Dragonflye is a fork of Shovill

**Comment by rpetit3** [comment]
At this point I'm not sure how to work with that Lint failure and will need some feedback

**Comment by rpetit3** [comment]
Was that the magic needed? Close and Reopen?

**Comment by Mxrcon** [comment]
yes! Very awkward :laughing: idk why but i tried to just reset the tests, not sure why this happened, maybe a bug with the test ci right? 
:rofl:happy that this pr will go foward.

File: modules/dragonflye/meta.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        pattern: "*fastq.gz"
```
Comment (grst) [inline modules/dragonflye/meta.yml:24]:
```suggestion
        pattern: "*.fastq.gz"
```

**Review by grst — APPROVED** [review]

**Comment by grst** [comment]
LGTM!

**Comment by rpetit3** [comment]
Thank you @Mxrcon for the [_"Did you try turning it off and on again?"_](https://www.google.com/search?q=did+you+try+turning+it+off+and+on+again) (Haha funny how that does it sometimes!)

@grst and Davi - For future reference, when I submit a PR and it gets out-of-date, should I update branch as needed or wait until the PR is ready for review?

**Comment by grst** [comment]
Feel free to update it whenever

**Comment by Mxrcon** [comment]
@rpetit3 I usually keep my pr's updated so the reviewers can merge it as soon as they let their review.

**Pull request merged**